cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/dataset_selector.spec.js

```diff
@@ -46,21 +46,7 @@ describe('dataset selector', { scrollBehavior: false }, () => {
 
   describe('select indices', () => {
     it('with SQL as default language', function () {
-      cy.getElementByTestId(`datasetSelectorButton`).click();
-      cy.getElementByTestId(`datasetSelectorAdvancedButton`).click();
-      cy.get(`[title="Indexes"]`).click();
-      cy.get(`[title=${DATASOURCE_NAME}]`).click();
-      cy.get(`[title="data_logs_small_time_1"]`).click(); // Updated to match loaded data
-      cy.getElementByTestId('datasetSelectorNext').click();
-
-      cy.get(`[class="euiModalHeader__title"]`).should('contain', 'Step 2: Configure data');
-
-      // Select SQL
-      cy.getElementByTestId('advancedSelectorLanguageSelect').select('OpenSearch SQL');
-      cy.getElementByTestId(`advancedSelectorTimeFieldSelect`).select('timestamp');
-      cy.getElementByTestId('advancedSelectorConfirmButton').click();
-
-      cy.waitForLoader(true);
+      cy.setIndexAsDataset('data_logs_small_time_1', DATASOURCE_NAME, 'OpenSearch SQL');
```

Comment:
Before this we should navigate to Discover. Although in the CICD and local the default page is discover and hence this might seem trivial. However when we run it in Neo the default page is workspaceHomePage which is not discover. Therefore adding a manual navigation to Discover page for the workspace will make it more robust.

Reply:
updated